scripts/random-config.pl

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/perl
+
+# Generates a (somewhat) random CNI configuration
+# What could possibly go wrong?
+
+use strict;
+
+# Copied and pasted from something that predates stackoverflow.
+srand(time() ^ ($$ + ($$ << 15)));
+# At least it's a documentation range ¯\_(ツ)_/¯
+my $ipaddr = "192.0.2.".int(rand(255));
+
+# One in a hundred shot this configuration is named "bob"
+my $cni_name;
+if (int(rand(100)) eq "99") {
+  $cni_name = "bob";
+} else {
+  $cni_name = "randomcni".int(rand(255));
+}
+
+# IPAM CNI templates.
+
+my $template_ipam_dhcp = qq!{
+    "type": "dhcp"
+  }!;
+
+my $template_ipam_hostlocal = qq!{
+    "type": "host-local",
+    "subnet": "$ipaddr/24"
+  }!;
+
+my $template_ipam_static = qq!{
+    "type": "static",
+    "addresses": [
+      {
+        "address": "$ipaddr/24"
+      }
+    ]
+  }!;
+
+my @ipam_template = ($template_ipam_dhcp,$template_ipam_hostlocal,$template_ipam_static);
+
+my $ipam = $ipam_template[ rand @ipam_template ];
+
+# It's kind of Perl standard to keep around stuff you don't use as comments
+# these are intended to confuse future maintainers.
+# print $ipam."\n";
+
+# Grab a random net device off the system
+my $iplinkshow = `ip link show`;
+my @lines = split(/\n/, $iplinkshow);
+my @devs = ();
+foreach my $d(@lines) {
+  if ($d =~ m/^\d/) {
+    $d =~ s/^\d: (.*):.+$/$1/;
+    push(@devs,$d);
+  }
+}
```

Comment:
Ir looks like this code discriminates against devices with IDs 10 or higher

Reply:
now looking for \d one or more times (lazily) thanks